### PR TITLE
feat: MKLDNN GRU forward/backward support

### PR DESCRIPTION
**Pull request by benjamim93**
## What changes were proposed in this pull request?
Enable MKLDNN GRU forward/backward

## How was this patch tested?
Unit tests. Compared MKLDNN results with BLAS results.

## Related links or issues (optional)
Jenkins passed: http://mlt-bddl01.sh.intel.com:8080/job/BigDL-PRVN-lite/916/

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/RNN.scala [hunk 1]
```diff
@@ -165,14 +171,6 @@ class RNN(
       PropKind.ForwardTraining
     }
 
-    /**
```
Comment (zhangxiaoli73) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/RNN.scala]: I think those comments should not be deleted.

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/RNN.scala [hunk 2]
```diff
@@ -181,8 +179,8 @@ class RNN(
         batchSize = inputs(0).shape(0)
         stepSize = inputs(0).shape(1)
       case _ =>
-        throw new UnsupportedOperationException("Not support such input format. " +
-          "The input format is: " + inputs(0).layout)
+        throw new UnsupportedOperationException("Not support such input format: " +
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/RNN.scala]: Unsupported input format

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/RNN.scala (hunk 1 above)
Comment (benjamim93) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/RNN.scala]: Changed back

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/RNN.scala (hunk 2 above)
Comment (benjamim93) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/RNN.scala]: Changed

**Pull request merged**